homeassistant/components/tessie/__init__.py

```diff
@@ -64,7 +74,45 @@ async def async_setup_entry(hass: HomeAssistant, entry: TessieConfigEntry) -> bo
         if vehicle["last_state"] is not None
     ]
 
-    entry.runtime_data = TessieData(vehicles=vehicles)
+    # Energy Sites
+    tessie = Tessie(session, api_key)
+    try:
+        products = (await tessie.products())["response"]
+    except TeslaFleetError as e:
+        raise ConfigEntryNotReady from e
+
+    energysites: list[TessieEnergyData] = []
+    for product in products:
+        if "energy_site_id" in product:
+            site_id = product["energy_site_id"]
+            api = EnergySpecific(tessie.energy, site_id)
+            energysites.append(
+                TessieEnergyData(
+                    api=api,
+                    id=site_id,
+                    live_coordinator=TessieEnergySiteLiveCoordinator(hass, api),
+                    info_coordinator=TessieEnergySiteInfoCoordinator(hass, api),
+                    device=DeviceInfo(
+                        identifiers={(DOMAIN, str(site_id))},
+                        manufacturer="Tesla",
+                        name=product.get("site_name", "Energy Site"),
+                    ),
+                )
+            )
+
+    # Populate coordinator data before forwarding to platforms
+    await asyncio.gather(
+        *(
+            energysite.live_coordinator.async_config_entry_first_refresh()
+            for energysite in energysites
+        ),
+        *(
+            energysite.info_coordinator.async_config_entry_first_refresh()
+            for energysite in energysites
+        ),
```

Comment:
I would imagine two of the same for loops are easily mergable right?

Reply:
I'm not sure what you mean, this creates a generator chain where each element is an awaitable that can be run simultaneously. Sure it could be a for loop with await but that would be slower.

Eventually it will also have a third generator for the vehicles.